src/test/java/seedu/address/model/amenity/AmenityTest.java

```diff
@@ -0,0 +1,48 @@
+package seedu.address.model.amenity;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class AmenityTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Amenity(null));
+    }
+
+    @Test
+    public void constructor_invalidAmenityType_throwsIllegalArgumentException() {
+        String invalidAmenityType = "";
+        assertThrows(IllegalArgumentException.class, () -> new Amenity(invalidAmenityType));
+    }
+
+    @Test
+    public void isValidAmenityType() {
+        // null amenity type
+        assertThrows(NullPointerException.class, () -> Amenity.isValidAmenityType(null));
+
+        // invalid amenity type
+        assertFalse(Amenity.isValidAmenityType("#wifi"));
```

Comment:
Same issue as `JsonAdaptedStudySpotTest.java` where it looks like we are testing for non-alphanumeric characters, when instead we are testing for Amenities not present in our `Amenity.VALID_TYPES` list.